include/abb_librws/rws_interface.h

```diff
@@ -287,6 +394,26 @@ class RWSInterface
    */
   std::string getIOSignal(const std::string& iosignal);
 
+  /**
+   * \brief A method for retrieving static information about a mechanical unit.
+   *
+   * \param mechunit for the mechanical unit's name.
+   * \param p_static_info for storing the retrieved information.
+   *
+   * \return bool indicating if the communication was successful or not. Note: No checks are made for "correct parsing".
+   */
+  bool getMechanicalUnitStaticInfo(const std::string& mechunit, MechanicalUnitStaticInfo* p_static_info);
```

Comment:
If these are your own methods, consider passing

```suggestion
  bool getMechanicalUnitStaticInfo(const std::string& mechunit, MechanicalUnitStaticInfo* const p_static_info);
```

or perhaps even:

```suggestion
  bool getMechanicalUnitStaticInfo(const std::string& mechunit, MechanicalUnitStaticInfo& p_static_info);
```

you're not passing this directly to some method/function not under our control which doesn't accept references, are you?

Reply:
And same comment below.

Reply:
Fixed in 44e45c4.

I opted for passing references, and I agree that it's high time to start moving away from passing raw pointers whenever it is suitable.

Reply:
I believe you mean fa396fd5b869675c9aebbd03b0c93338b5c0306d, correct?

Otherwise: nice :+1:

Reply:
Correct